awwvision/src/main/java/com/google/cloud/servicebroker/samples/awwvision/controller/RedditScraperController.java

```diff
@@ -88,11 +89,11 @@ void storeAndLabel(RedditResponse response) {
         }
         try {
           // Only label and upload the image if it does not already exist in storage.
-          StorageObject existing = storageApi.get(listing.data.url);
+          StorageObject existing = cuteImageService.get(listing.data.url);
           if (existing == null) {
-            String label = visionApi.labelImage(raw);
+            String label = imageLabelingService.labelImage(raw);
             if (label != null) {
-              storageApi.uploadJpeg(listing.data.url, url, ImmutableMap.of("label", label));
+              cuteImageService.uploadJpeg(listing.data.url, url, ImmutableMap.of("label", label));
             }
```

Comment:
what do you think about refactoring this to this? I think it'll make it cleaner, and reduce the useless work we do on our outbound network calls:

```
void storeAndLabel(RedditResponse response) {
  for (Listing listing : response.data.children) {
    storeAndLabel(listing); // functional support?
  }
}

private void storeAndLabel(final Listing listing) {
  Objects.requireNonNull(listing, "listing must not be null");
  if (listing.data.preview == null) { // is this a bug that should be listing.data.url?
    return;
  }
  
  final String dataUrl = listing.data.url;
  try {
    // Only label and upload the image if it does not already exist in storage.
    final StorageObject existing = cuteImageService.get(dataUrl);
    if (existing != null) {
      return;
    }
    
    final URL url = new URL(dataUrl);
    final byte[] raw = download(url);
    final String label = imageLabelingService.labelImage(raw);
    if (label != null) {
      cuteImageService.uploadJpeg(dataUrl, url, ImmutableMap.of("label", label));
    }
  } catch (IOException ex) {
    logger.error("Issue with labeling or uploading image " + dataUrl, ex);
  }
}
```